packages/twenty-server/src/core/auth/services/token.service.ts

```diff
@@ -232,6 +236,31 @@ export class TokenService {
     };
   }
 
+  async generateSwitchWorkspaceToken(
+    accessToken: string,
+    workspaceId: string,
+  ): Promise<Verify> {
+    const secret = this.environmentService.getRefreshTokenSecret();
+    const jwtPayload = await this.verifyJwt(accessToken, secret);
```

Comment:
verification is already done by the `@UseGuards(JwtAuthGuard)` in `generateJWT`
Think you can just pass the workspaceId and the authenticated user as parameters, and make sure the user belongs to the workspaceId before generating a new token

Reply:
Using `@AuthUser`